Assets/Scripts/DataModels/DataModelsUtils.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using UnityEngine;
+
+public static class DataModelsUtils
+{
+    public const string FILE_EXTENSION = ".truckin";
+
+    public static async void SaveFileAsync<T>(string fileName, string folderName, T dataModel)
+    {
+        string folderPath = GetSaveFolderPath(folderName);
+        if (!Directory.Exists(folderPath))
+        {
+            try
+            {
+                Directory.CreateDirectory(folderPath);
+            }
+            catch (Exception e)
+            {
+                Debug.LogError($"{e.Message}\n{e.StackTrace}");
+            }
+        }
+
+        string filePath = Path.Combine(folderPath, fileName + FILE_EXTENSION);
+        string fileContents = JsonUtility.ToJson(dataModel);
+
+        try
+        {
+            var buffer = Encoding.UTF8.GetBytes(fileContents);
+            
+            using (var fileStream = new FileStream(filePath, FileMode.OpenOrCreate,
+                FileAccess.Write, FileShare.None, buffer.Length, true))
+            {
+                await fileStream.WriteAsync(buffer, 0, buffer.Length);
+            }
+        }
+        catch (Exception e)
+        {
+            Debug.LogError($"{e.Message}\n{e.StackTrace}");
+        }
+    }
+
+    public static async Task<T> LoadFileAsync<T>(string fileName, string folderName) where T : class, new()
+    {
+        Debug.Log($"{typeof(T)} load commencing.");
+        string folderPath = GetSaveFolderPath(folderName);
+        string filePath = Path.Combine(folderPath, fileName + FILE_EXTENSION);
+
+        if (File.Exists(filePath))
+        {
+            try
+            {
+                string json;

```

Comment:
I was getting json parse errors invalid value with the code we did last time. 
This seems to work fine. And it is less lines so why not. I don't think there's any downside.

Reply:
yeah this should be fine